File: app/src/main/java/com/unibs/zanotti/inforinvestigador/data/model/ExpandableCommentGroup.java
package com.unibs.zanotti.inforinvestigador.data.model;

import com.xwray.groupie.ExpandableGroup;

public class ExpandableCommentGroup extends ExpandableGroup {
    public ExpandableCommentGroup(Comment comment, int depth) {
        super(new ExpandableCommentItem(comment, depth));

        for (Comment comm : comment.getChildren()) {
            add(new ExpandableCommentGroup(comm, depth));
            depth++;
        }
    }
}

File: app/src/main/java/com/unibs/zanotti/inforinvestigador/data/ExpandableCommentGroup.kt
package com.unibs.zanotti.inforinvestigador.data

import com.xwray.groupie.ExpandableGroup

class ExpandableCommentGroup constructor(comment: Comment, depth: Int = 0) :
    ExpandableGroup(ExpandableCommentItem(comment, depth)) {

    init {
        for (comm in comment.children) {
            add(ExpandableCommentGroup(comm, depth.plus(1)))
        }
    }
}